gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -280,20 +290,22 @@ protected void startUp() {
    * Note this should only be called from the {@link Orchestrator} or {@link org.apache.gobblin.service.monitoring.DagActionStoreChangeMonitor}
    */
   public synchronized void addDag(Dag<JobExecutionPlan> dag, boolean persist, boolean setStatus) throws IOException {
-    if (persist) {
-      //Persist the dag
-      this.dagStateStore.writeCheckpoint(dag);
-    }
-    int queueId = DagManagerUtils.getDagQueueId(dag, this.numThreads);
-    // Add the dag to the specific queue determined by flowExecutionId
-    // Flow cancellation request has to be forwarded to the same DagManagerThread where the
-    // flow create request was forwarded. This is because Azkaban Exec Id is stored in the DagNode of the
-    // specific DagManagerThread queue
-    if (!this.runQueue[queueId].offer(dag)) {
-      throw new IOException("Could not add dag" + DagManagerUtils.generateDagId(dag) + "to queue");
-    }
-    if (setStatus) {
-      submitEventsAndSetStatus(dag);
+    if (isActive) {
```

Comment:
Stylistic nit but sometimes I like doing the reverse and exit early. Exactly the same thing but sometimes more readable than a lot of nested if statements